interop/xds/client/client.go

```diff
@@ -368,6 +378,22 @@ func parseRPCMetadata(rpcMetadataStr string, rpcs []string) []*rpcConfig {
 
 func main() {
 	flag.Parse()
+	if *enableCSMObservability {
+		exporter, err := prometheus.New()
+		if err != nil {
+			logger.Fatalf("Failed to start prometheus exporter: %v", err)
+		}
+		provider := metric.NewMeterProvider(
+			metric.WithReader(exporter),
+		)
+		go http.ListenAndServe(":9464", promhttp.Handler())
```

Comment:
Should we do something like `go func() { if err := ListenAndServe; err != nil { logger.Fatalf } }()`, so if the port is taken we get a noisy error instead of just things not working right?

Reply:
Done for both client and server.